### ISSUE TITLE
bug in pycomment

### ISSUE THREAD

**Issue by pfuntner** [pfuntner/toys#33]
While working on the documentation for `pycomment`, I noticed a bug.  Consider this edit buffer:

```
      1 #! /usr/bin/env python
      2 
      3 import sys
      4 
      5 print 'Before `if`'
      6 if len(sys.argv) > 1:
      7   """
      8     print the arguments
      9   """
     10   # here we go
     11   for num in range(1, len(sys.argv)):
     12     # this is a naive way to print a list
     13     print sys.argv[num]
     14   # # this is a better way to print a list
     15   # print '\n'.join(sys.argv[1:])
```
If you use command `:11,15!pycomment`, you'll end up with:
```
      1 #! /usr/bin/env python
      2 
      3 import sys
      4 
      5 print 'Before `if`'
      6 if len(sys.argv) > 1:
      7   """
      8     print the arguments
      9   """
     10   # here we go
     11   # for num in range(1, len(sys.argv)):
     12   #   # this is a naive way to print a list
     13   #   print sys.argv[num]
     14   # this is a better way to print a list
     15   print '\n'.join(sys.argv[1:])
```
I think this is what I wanted but if you run the same command again, you should end up with the original edit buffer but it doesn't work that way:
```
      1 #! /usr/bin/env python
      2 
      3 import sys
      4 
      5 print 'Before `if`'
      6 if len(sys.argv) > 1:
      7   """
      8     print the arguments
      9   """
     10   # here we go
     11   for num in range(1, len(sys.argv)):
     12     # this is a naive way to print a list
     13     print sys.argv[num]
     14   this is a better way to print a list
     15   # print '\n'.join(sys.argv[1:])
```
Line 14 is incorrect.  It should be commented out.

**Comment by pfuntner** [comment]
`pycomment` was inspired by the `Ctrl-/` command of PyCharm and I think I just had a misconception about how it worked.  If I start with the same original file in PyCharm and use `Ctrl-/` on the same lines, I get:

```
      1 #! /usr/bin/env python
      2
      3 import sys
      4
      5 print 'Before `if`'
      6 if len(sys.argv) > 1:
      7  """
      8     print the arguments
      9   """
     10   # here we go
     11   # for num in range(1, len(sys.argv)):
     12   #   # this is a naive way to print a list
     13   #   print sys.argv[num]
     14   # # # this is a better way to print a list
     15   # # print '\n'.join(sys.argv[1:])
```
I had a real misconception about the behavior and I think I might have been modeling the behavior from memory.  From this behavior, I would say:

  - if all lines are commented out, they will be uncommented (at least the first level, if there are multiple level of comments)
  - if **not** all lines are comments out, they **_will be_**

**Comment by pfuntner** [comment]
I fixed this with #34
